### PR TITLE
Add default timeout support to DjangoCache

### PR DESCRIPTION
**Pull request by mverteuil**
I missed this when I implemented the class, so it's more of a bugfix.

### CODE REVIEW

File: gapipy/cache.py [hunk 1]
```diff
@@ -200,6 +201,8 @@ def get(self, key):
             return self.client.get(key)
 
         def set(self, key, value, timeout=None):
+            if timeout is None:
+                timeout = self.default_timeout
```
Comment (jonprindiville) [inline gapipy/cache.py:205]: 👍 yeah, this works for me

My alternate theory was "don't supply a timeout value to the set call" (like `self.client.set(key, value)`) and then we'd get whatever our Django settings say is the default timeout for that cache backend.

The approach in this PR is nice, though, in that DjangoCache becomes a bit more like how the other gapipy cache backends work (and for our particular use-case we'll add a `'cache_options': { 'default_timeout': ..., }` back into our configs)

Comment (mverteuil) [reply gapipy/cache.py:205]: Yeah, I went with "consistent with gapipy caches" > "django has a special way"

Briefly considered making constructor args to choose... but I decided not to go further than I had to for now

**Review by marz619 — APPROVED** [review]
👍🏽  LGTM!

**Comment by jonprindiville** [comment]
Oh yeah, for context: prior to the changes in this PR, if you were using the DjangoCache you would have been getting cache entries that persist forever because we had been passing `None` as the default timeout

**Pull request merged**